### PR TITLE
feat: Make LostPixel compatible with StoryBook V8 and upwards

### PR DESCRIPTION
**Pull request by davincho**
Starting with Storybook >= v7, the API for retrieving stories has changed - see https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#storystore-and-methods-deprecated

This PR makes LP ready for V8 and V9 while still keeping backwards compatibility for V6.

`window.__STORYBOOK_PREVIEW__` has a `ready` promise in V8 or above and is used as a signal which SB version we are trying to crawl.
Based on this we either await on `window.__STORYBOOK_PREVIEW__.ready` or on `window.__STORYBOOK_CLIENT_API__` and call `window.__STORYBOOK_CLIENT_API__.storyStore.cacheAllCSFFiles` (accessing the StoryStore is deprecated and will be removed in V9).

For the extraction part we use `await window.__STORYBOOK_PREVIEW__.extract` - I checked the existence of this method in V6, V7 and V8 ✅ 
V6.4 still needs a fallback of using `window.__STORYBOOK_CLIENT_API__.raw`

 All Submissions:

* [ ] Have you followed the guidelines in our [Contributing document](https://github.com/lost-pixel/lost-pixel/blob/main/CONTRIBUTING.md)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/lost-pixel/lost-pixel/pulls) for the same update/change?

### New Feature Submissions:

1. [ ] Was the new feature discussed with core maintainers?
2. [ ] Does your submission pass tests?
3. [ ] Have you lint your code locally before submission?
4. [ ] Does the change require an update in documentation? Have you updated the documentation in case it's true?

### Changes to Core Features:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?
* [ ] Does the change require an update in documentation? Have you updated the documentation in case it's true?

### CODE REVIEW

**Comment by davincho** [comment]
Something to keep in mind though: Apparently in SB version 9 access to the storystore will be deprecated as well 🤔 

https://github.com/storybookjs/storybook/blob/670c58c98331edc764efb72e2bfcebcbc8f53580/code/lib/preview-api/src/modules/preview-web/Preview.tsx#L91-L95

**Review by d-ivashchuk — APPROVED** [review]

File: src/crawler/storybook.ts [hunk 1]
```diff
@@ -99,95 +105,124 @@ export const collectStoriesViaWindowApi = async (
...
+  if (isV8OrAbove) {
+    // SB v8 and above
+    await page.evaluate(async () => {
+      const { __STORYBOOK_PREVIEW__: api } = window as WindowObject;
 
-        const fetchStories = () => {
-          const { __STORYBOOK_CLIENT_API__: api } = window as WindowObject;
+      if (api.ready) {
+        await api.ready();
+      }
```
Comment (chriskalmar) [inline src/crawler/storybook.ts:129]: nice find 👌

File: src/crawler/__snapshots__/storybook.spec.ts.snap [hunk 2]
```diff
@@ -1671,3 +1867,451 @@ exports[`collectStories should collect stories from StoryBook: ViaWindowApi Stor
...
+    await expect(loginButton).toBeInTheDocument();
+    await userEvent.click(loginButton);
+    await expect(loginButton).not.toBeInTheDocument();
+    const logoutButton = canvas.getByRole('button', {
+      name: /Log out/i
+    });
+    await expect(logoutButton).toBeInTheDocument();
+  }
+}",
+          },
```
Comment (chriskalmar) [inline src/crawler/__snapshots__/storybook.spec.ts.snap:2303]: Interesting. 
we might include this in the UI one day.

**Review by chriskalmar — APPROVED** [review]
Thank you for your contribution @davincho 🙏 
Awesome job ❤️

**Comment by chriskalmar** [comment]
Released in https://github.com/lost-pixel/lost-pixel/releases/tag/v3.16.0

**Pull request merged**